### DISCUSSION TITLE
Micropython PICO memory management

### DISCUSSION THREAD

**Discussion by mastraa** [micropython/micropython#14060] [Using MicroPython]
Dear all,

I need some holep with PICO memory management using MicroPython.
I can't understand something:

I'm using gc.mem_alloc(), gc.mem_free() and gc.collect() toi understand how my action will influence memory usage. There's something I cannot understand.

If I call del function to delete variable no changes in memory allocated and free seems to happen. Even using gc.collect() function in between.

How do they work and how I can clean some memory between two loop for example?

PS: i get a general Memory Error even if I think memory is quite free...What could it be?

Thank you for helping
Andrea

**Comment by mendenm** [comment]
If you are doing this via Thonny, the results are essentially random.  Something inside the Thonny communications messes these numbers up.  I got bitten by this as one of my earliest  encounters with mpy on Pico.  If you aren't using Thonny, you might want to give more specific examples.

**Reply by mastraa** [reply]
dear mendenm, I'm using Thonny so it's "clear". i cannot know how to menage it.

However using a gc.collect() seems to solve it, maybe it load too much things at the beginning and he could remove something later...

Thank you,
Andrea
